src/main/java/uk/ac/cam/cl/dtg/segue/auth/SegueLocalAuthenticator.java

```diff
@@ -244,8 +244,13 @@ public void ensureValidPassword(final String password) throws InvalidPasswordExc
             throw new InvalidPasswordException("Invalid password. You cannot have an empty password.");
         }
 
-        if (password.length() < 6) {
-            throw new InvalidPasswordException("Password must be at least 6 characters in length.");
+        if (password.length() < 12) {
+            throw new InvalidPasswordException("Password must be at least 12 characters in length.");
+        }
+
+        //  Special characters: !"#$%&'()*+,-./:;<=>?@[\]^_`{|}~
```

Comment:
This comment is not very clear: does it mean that these are valid special characters or not? I think it means that they are allowed, but in any case, I would completely remove this comment and write a parametrised test for each one of these characters.

Reply:
Ah sorry, that was a note for myself while I was sorting out which characters need escaping. It can go now.

Reply:
Unresolving this conversation as the tests for this are still missing. You have a nice list with all the accepted special characters, so you could have a parametrised test to ensure that these characters are accepted as valid.